From 2dbae2d1da3c39606c5021e644841bfd6bcb0b44 Mon Sep 17 00:00:00 2001
From: arition <aritionkb@gmail.com>
Date: Thu, 27 Aug 2020 18:22:59 -0700
Subject: [PATCH] cmake patch

---
 CMakeLists.txt | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fa50f15..69d9984 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,6 +8,8 @@ option(WITH_CUDA "Enable CUDA support" OFF)
 if(WITH_CUDA)
   enable_language(CUDA)
   add_definitions(-D__CUDA_NO_HALF_OPERATORS__)
+  message(STATUS "Torchvision CUDA enabled")
+  add_definitions(-DWITH_CUDA)
 endif()
 
 find_package(Python3 COMPONENTS Development)
@@ -25,9 +27,7 @@ add_library(${PROJECT_NAME} SHARED ${MODELS_SOURCES} ${OPERATOR_SOURCES})
 target_link_libraries(${PROJECT_NAME} PRIVATE ${TORCH_LIBRARIES} Python3::Python)
 set_target_properties(${PROJECT_NAME} PROPERTIES EXPORT_NAME TorchVision)
 
-target_include_directories(${PROJECT_NAME} INTERFACE
-  $<BUILD_INTERFACE:${HEADERS}>
-  $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
+include_directories(torchvision/csrc)
 
 include(GNUInstallDirs)
 include(CMakePackageConfigHelpers)
@@ -65,3 +65,6 @@ if(WITH_CUDA)
     DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME}/cuda)
 endif()
 install(FILES ${MODELS_HEADERS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME}/models)
+
+set_property(TARGET torch_cuda PROPERTY INTERFACE_COMPILE_OPTIONS "")  
+set_property(TARGET torch_cpu PROPERTY INTERFACE_COMPILE_OPTIONS "")
-- 
2.27.0.windows.1

